gapis/gfxapi/gfxapi.proto

```diff
@@ -102,6 +92,26 @@ enum UniformType {
 	Double = 4;
 }
 
+// ResourceData represents the resource state at a single point in a capture
+message ResourceData {
+	oneof data {
+		Texture texture = 1;
+		Shader shader = 2;
+		Program program = 3;
+	}
+}
+
+// Texture represents a texture resource.
+message Texture {
+	oneof type {
```

Comment:
> Are you proposing that a 3D texture is just N x image.Info2D?
Yes.  As far as I am concerted 3D texture and 2D texture array are interchangeable, including the UI visualization.

> My personal opinion is that the server interface should be explicit
I am in favor of explict, however what we are doing here is to create special type for the case when array length is 1.  I would prefer less copy-pasted code.

> This is purely my opinion though, so I'm happy to open this up for a vote.
Fair enough, @pmuetschard @AWoloszyn ... opinions?